src/api/app/controllers/trigger_controller.rb

```diff
@@ -1,16 +1,71 @@
 class TriggerController < ApplicationController
+  validate_action rebuild: { method: :post, response: :status }
+  validate_action release: { method: :post, response: :status }
   validate_action runservice: { method: :post, response: :status }
 
+  before_action :require_valid_token
+
   #
   # This controller is checking permission always only on the base of tokens
   #
   skip_before_action :extract_user
   skip_before_action :require_login
 
-  # github.com sends a hash payload
-  skip_before_action :validate_params, only: [:runservice]
+  def rebuild
+    Backend::Api::Sources::Package.rebuild(@pkg.project.name, @pkg.name)
+    render_ok
+  end
+
+  # rubocop:disable Lint/NonLocalExitFromIterator
```

Comment:
why? :sob:

Reply:
because we must escape from the loops with proper error message and don't want to have additional checking code just to get away with break.

Reply:
ok